### PR TITLE
[MetaSearch] Save Settings

### PR DESCRIPTION
**Pull request by frida-161**
This PR allows the MetaSearch Plugin to save its Settings (using QgsSettings).

The plugin already did use the QgsSettings module to some extent but not all the Settings in the Settings tab were saved to QGIS Settings and loaded from there on startup.
This PR implements saving (on change) and loading(on startup) of the `timeout`, `maxrecords` and `disable_ssl_verification` Settings.

### CODE REVIEW

**Comment by towa** [comment]
There also appears to be a broken reference in the code.
`apidialog.py` references `apidialog.ui` even though the UI file is called `xmldialog.ui`

**Comment by agiudiceandrea** [comment]
> There also appears to be a broken reference in the code.
> `apidialog.py` references `apidialog.ui` even though the UI file is called `xmldialog.ui`

See https://github.com/qgis/QGIS/pull/46292.

**Comment by towa** [comment]
Seems like I missed that PR.
I merged master in my branch and changed the reference again, so it is now referenced as apidialog.ui

File: python/plugins/MetaSearch/dialogs/maindialog.py [hunk 1]
```diff
@@ -88,10 +88,12 @@ def __init__(self, iface):
...
         self.startfrom = 1
-        self.maxrecords = 10
-        self.timeout = 10
-        self.disable_ssl_verification = False
         self.constraints = []
+        self.maxrecords = int(self.settings.value('/MetaSearch/returnRecords', 10))
+        self.timeout = int(self.settings.value('/MetaSearch/timeout', 10))
+        # QgsSettings.value returns 'false' not False(!)
+        self.disable_ssl_verification = self.settings.value(
+            '/MetaSearch/disableSSL', 'false') != 'false'
```
Comment (nyalldawson) [inline python/plugins/MetaSearch/dialogs/maindialog.py:96]: Try:

```suggestion
        self.disable_ssl_verification = self.settings.value(
            '/MetaSearch/disableSSL', False, bool)
```

File: python/plugins/MetaSearch/dialogs/maindialog.py [hunk 2]
```diff
@@ -139,12 +141,30 @@ def __init__(self, iface):
...
-        self.spnRecords.setValue(
-            int(self.settings.value('/MetaSearch/returnRecords', 10)))
+
+        # load settings
+        self.spnRecords.setValue(self.maxrecords)
+        self.spnRecords.valueChanged.connect(_on_records_change)
+        self.spnTimeout.setValue(self.timeout)
+        self.spnTimeout.valueChanged.connect(_on_timeout_change)
+        self.disableSSLVerification.setCheckState(
+            int(self.disable_ssl_verification) * 2)
```
Comment (nyalldawson) [inline python/plugins/MetaSearch/dialogs/maindialog.py:166]:
```suggestion
        self.disableSSLVerification.setChecked(self.disable_ssl_verification)
```

**Comment by nyalldawson** [comment]
@tomkralidis is this ok for you?

**Comment by tomkralidis** [comment]
+1, thanks for the contribution @towa !

**Review by tomkralidis — APPROVED** [review]

File: python/plugins/MetaSearch/dialogs/maindialog.py [hunk 3]
```diff
@@ -88,10 +88,12 @@ def __init__(self, iface):
 
         # form inputs
         self.startfrom = 1
-        self.maxrecords = 10
-        self.timeout = 10
-        self.disable_ssl_verification = False
         self.constraints = []
+        self.maxrecords = int(self.settings.value('/MetaSearch/returnRecords', 10))
+        self.timeout = int(self.settings.value('/MetaSearch/timeout', 10))
+        # QgsSettings.value returns 'false' not False(!)
```
Comment (nyalldawson) [inline python/plugins/MetaSearch/dialogs/maindialog.py:94]:
```suggestion
```

**Pull request merged**